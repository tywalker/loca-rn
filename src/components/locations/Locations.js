import React, { Component } from 'react';
import { FlatList, Image, StyleSheet, Text, View } from 'react-native';
import { connect } from 'react-redux';
import { getBoundingBoxFromGeo } from '../../services/api';

import { locationsRequest } from '../../actions/locations';

class Locations extends Component {

  componentDidUpdate(prevProps) {
    const { dispatch, lat, lon, places } = this.props;

    if (prevProps.lat !== lat && prevProps.lon !== lon) {
      let bbox = getBoundingBoxFromGeo(lat, lon, 5.0);

      dispatch(locationsRequest(bbox, 5.0));
    }
  }

  _keyExtractor = ( item, index ) => item.id;

  _renderItem = ({ item, index }) => (
    <View
      id={ index }
      style={ styles.location }
    >
      <Text style={ styles.locaLabel }>{ item.woe.name }</Text>
      <View style={ styles.locaImageContainer }>
        <Image
          source={{ uri: "https://placeimg.com/200/150/any" }}
          style={{ height: 150, width: 200 }}
        />
      </View>
    </View>
  );


  render() {
    const { places } = this.props;

    return (
      <View style={styles.container}>
        <View style={ styles.topNavigation }></View>
        <View style={ styles.mainViewContainer }>

          <FlatList
            data={ places }
            keyExtractor={ this._keyExtractor }
            renderItem={ this._renderItem }
          />

        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#F5FCFF',
  },
  topNavigation: {
    flex: 0.2,
    backgroundColor: '#f0f0f0'
  },
  mainViewContainer: {
    flex: 0.8,
    backgroundColor: '#f8f8f8',
    alignItems: 'center'
  },
  locationsContainer: {
    flex: 1,
    minWidth: '90%',
    maxWidth: '90%',
    backgroundColor: '#f9f9f9'
  },
  location: {
    height: 300,
  },
  locaLabel: {
    color: "#555",
    fontSize: 16
  },
  locaImageContainer: {
    minHeight: 150,
    minWidth: 200
  }
});

function mapStateToProps(state) {
  return {
    places: state.locations.places,
    lat: state.locations.lat,
    lon: state.locations.lon
  }
}

export default connect(mapStateToProps)(Locations);
