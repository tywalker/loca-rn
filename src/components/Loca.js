import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';

import Locations from './locations/Locations';

class Loca extends Component {
  render() {
    return (
      <View style={ styles.container }>
        <Locations />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#F5FCFF',
  },
});

export default Loca;
