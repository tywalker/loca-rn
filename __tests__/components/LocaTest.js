import React from 'react';
import { View } from 'react-native';
import Locations from '../../src/components/locations/locations';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../../src/reducers/locations';
import Loca from '../../src/components/Loca';

const store = createStore(
  rootReducer,
  applyMiddleware(thunk),
);

describe("App loaded", () => {
  it('renders Loca component', () => {
    const wrapper = shallow(
      <Provider store={ store }>
        <Loca/>
      </Provider>
    );

    expect(wrapper).toBeTruthy();
  });

  it('defaults view state to loading', () => {
    const state = store.getState();

    expect(state.views.view).toBe('loading');
  });

  it('should successfully render the Locations component view', () => {
    let styles = {
      container: {}
    }

    const wrapper = shallow(
      <Loca store={store}>
        <View style={ styles.container }>
          <Locations boundingBox={ this.bbox } />
        </View>
      </Loca> 
    );

    expect(wrapper.dive()).toMatchSnapshot();
  });
});
