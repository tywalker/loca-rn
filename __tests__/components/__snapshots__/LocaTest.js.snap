// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App loaded should successfully render the Locations component view 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loca
    dispatch={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <Component
      style={Object {}}
    >
      <Connect(Locations) />
    </Component>
  </Loca>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(Locations) />,
      "style": Object {
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "boundingBox": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(Locations) />,
        "style": Object {
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "boundingBox": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
